packages/react-lowcode/src/localization/json.ts

```diff
@@ -27,16 +41,51 @@ export function parseLocaleAST(ast: SourceFile, languageLocale = "en") {
 }
 
 // it preserves original JSON formatting
-export function patchLocaleJSON(localeFile: string, changedMessages: Message[], originalMessages: Message[]) {
+export function patchLocaleJSON(
+  localeFile: string,
+  changedMessages: Message[],
+  originalMessages: Message[]
+) {
   for (let i = changedMessages.length; i >= 0; i--) {
     if (changedMessages[i]?.value == originalMessages[i]?.value) {
       console.log("Equal")
     } else {
-      const before = localeFile.substring(0, originalMessages[i].position.pos + 1)
+      const before = localeFile.substring(
+        0,
+        originalMessages[i].position.pos + 1
+      )
       const after = localeFile.substring(originalMessages[i].position.end - 1)
       localeFile = before + changedMessages[i].value + after
     }
   }
 
   return localeFile
 }
+
+export function putLocaleMessage(
+  messageId: string,
+  newValue: string,
+  originalLocaleStringJSON: string | undefined
+) {
+  const originalMessages =
+    parseLocaleJSON(originalLocaleStringJSON as string) || []
+  const found = originalMessages?.find((message) => message.id == messageId)
+
+  if (originalMessages?.length == 0) {
```

Comment:
@chvister, nice work. Just last few commetns:

please move/extract following logic to a function `newFunction(changedMessage: Message,  originalMessages: Message[])` and use it in both existing functions:
* putMessage
* patchLocaleJSON (for backward compatibility)